agent/consul/leader_metrics.go

```diff
@@ -151,3 +152,17 @@ func initLeaderMetrics() {
 		metrics.SetGaugeWithLabels(g.Name, float32(math.NaN()), g.ConstLabels)
 	}
 }
+
+// shouldWarn checks to see if we are close enough to the cert expiring that we
+// should send out a WARN log message.
+// It defaults to returning true if the cert will expire within 28 days or 40%
+// of the certs total duration, whichever is shorter.
+func expiresSoon(before, after time.Duration) bool {
+	warningPeriod := 28 * (24 * time.Hour) // 28 days (default)
+	seventyDays := 70 * (24 * time.Hour)   // 28 days is 40% of 70 days
+	totalDuration := time.Duration(before + after)
+	if totalDuration < seventyDays {
+		warningPeriod = (totalDuration * 4) / 10 // 40% of total duration
+	}
+	return after < warningPeriod
+}
```

Comment:
In the current implementation, the 40% assumption is baked into two lines of code rather than defined in one place. Here's my attempt at reworking it slightly to define in one place (including naming changes from my other comment).

```suggestion
func expiresSoon(timeSinceLifetimeStart, timeUntilLifetimeEnd time.Duration) bool {
	warningPeriod = 28 * (24 * time.Hour) // 28 days (default) 
	
	totalLifetime := time.Duration(timeSinceLifetimeStart + timeUntilLifetimeEnd)
	warningPeriodByTotalLifetime = (totalLifetime * 4) / 10 // 40% of total duration
	if warningPeriodByTotalLifetime < warningPeriod {
		warningPeriod = warningPeriodByTotalLifetime 
	}
	
	return timeUntilLifetimeEnd < warningPeriod
```

I defer to you/eng on whether to make sure a change / whether it helps with clarity!

Reply:
I see what you're shooting for, trying to make the 2 periods more explicit. I'll think about this one.